src/tsdb.c

```diff
@@ -132,6 +174,7 @@ int SeriesItertorGetNext(SeriesItertor *iterator, Sample *currentSample) {
             break;
         } else {
             memcpy(currentSample, &internalSample, sizeof(Sample));
+            iterator->currentSampleIndex++;
```

Comment:
@danni-m , please note this change

Reply:
@maital 
Its my fault that `currentSampleIndex` still exists, but this is a leftover from a previous iterator implementation.
I'd prefer to remove it, and let the iterator user to count the number of samples.
Is it possible for you?